test/lib/Analysis/TestAlias.cpp

```diff
@@ -57,17 +59,15 @@ struct TestAliasPass
     auto getAllocOpNames = [&](Value value) {
       dataflow::Lattice<AliasInfo> *latticeElement =
           analysis->getLatticeElement(value);
-      SmallVector<std::string, 4> opNames;
+      std::set<std::string> opNames;
       if (latticeElement) {
         auto &info = latticeElement->getValue();
         for (auto &alias : info.getAllocs()) {
           auto opName =
               getValueOperandName(alias.getDefiningOp()->getResult(0), state);
-          opNames.push_back(std::move(opName));
+          opNames.insert(std::move(opName));
```

Comment:
The change might break mlir tests.

https://github.com/openai/triton/blob/78c28bf5f6049d3f224b22eba1be0db59706a58c/test/Analysis/test-alias.mlir#L133

`%0` could be defined in either branch of `scf.if`, so we didn't remove duplicate entries here.

Reply:
> The change might break mlir tests.

Are you saying we **want** to see the duplication because it's important for what is tested? (I didn't try to understand the test, this is why I highlighted this)

I'll revert this and loop you in for a case of a new duplicated entry that is generated now.

Reply:
Sure

Reply:
I reverted that and I don't see any lit failures. @joker-eph did you change that due to some failures?